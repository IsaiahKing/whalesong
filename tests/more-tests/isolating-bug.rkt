#lang planet dyoo/whalesong/base
(require (for-syntax racket/base
                     syntax/struct))


(define-syntax (my-define-struct stx)
  (syntax-case stx ()
    [(_ name (fields ...) kw ...)
     (with-syntax ([(names ...)
                    (build-struct-names #'name
                                        (syntax->list #'(fields ...))
                                        #f
                                        #f)])
       (with-syntax ([cnstr (syntax-case #'(names ...) ()
                              [(struct:name-id constructor misc ...)
                               #'constructor])])
         #'(begin
             (define-values (names ...)
               (let ()
                 (begin
                   (define-struct name (fields ...) kw ...)
                   (let ([cnstr (lambda args
                                  (apply cnstr args))])
                     (values names ...))))))))]))

(my-define-struct swf (f) #:mutable)
make-swf
swf?
swf-f
