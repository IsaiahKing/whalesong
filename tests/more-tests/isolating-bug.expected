#<procedure:make-swf>
#<procedure:swf?>
#<procedure:swf-f>
